kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -257,11 +324,13 @@ object Channel:
         def init[A](
             _capacity: Int,
             access: Access = Access.MultiProducerMultiConsumer
-        )(using initFrame: Frame, allow: AllowUnsafe): Unsafe[A] =
+        )(using putType: PutType[A], initFrame: Frame, allow: AllowUnsafe): Unsafe[A] =
+            import putType.Put
+
             new Unsafe[A]:
                 val queue = Queue.Unsafe.init[A](_capacity, access)
                 val takes = new MpmcUnboundedXaddArrayQueue[Promise.Unsafe[Closed, A]](8)
-                val puts  = new MpmcUnboundedXaddArrayQueue[(A, Promise.Unsafe[Closed, Unit])](8)
+                val puts  = new MpmcUnboundedXaddArrayQueue[(Put, Promise.Unsafe[Closed, Unit])](8)
```

Comment:
What if we keep the current type for unbatched puts, then swap the order for batched puts?

`(A, Promise.Unsafe[...]) | (Promise.Unsafe[...], Chunk[A])`

This should have the same cost as we need to match on the type anyways, but we can avoid a wrapper.

Reply:
Ah good point. Only potential problem is edge case where `A =:= Promise.Unsafe[?,?]`.

How about we replace the tuple with an `enum`?:
```scala
enum Put[A]:
  case Value(value: A, promise: Promise.Unsafe[Closed, Unit])
  case Batch(batch: Chunk[A], promise: PromiseUnsafe[Closed, Unit])
```

I take it case classes and tuples have the same overhead?